clients/cli/src/main/java/org/apache/gravitino/cli/GravitinoCommandLine.java

```diff
@@ -318,14 +319,29 @@ private void handleSchemaCommand() {
     String catalog = name.getCatalogName();
 
     Command.setAuthenticationMode(auth, userName);
+    List<String> missingEntities = Lists.newArrayList();
+    if (metalake == null) missingEntities.add(CommandEntities.METALAKE);
+    if (catalog == null) missingEntities.add(CommandEntities.CATALOG);
 
     // Handle the CommandActions.LIST action separately as it doesn't use `schema`
     if (CommandActions.LIST.equals(command)) {
+      if (!missingEntities.isEmpty()) {
+        System.err.println("Missing required argument(s): " + COMMA_JOINER.join(missingEntities));
+        Main.exit(-1);
+      }
       newListSchema(url, ignore, metalake, catalog).handle();
       return;
     }
```

Comment:
Can we move this handling of LIST command "back" to the switch case clause below?

Reply:
Hi @tengqm , I’ve finished updating the code. Please take a look at the PR again when you have time.